app/src/main/java/com/github/polypoly/app/network/storable/StorableObject.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.polypoly.app.network.storable
+
+import kotlin.reflect.KClass
+
+/**
+ * [StorableObject] is the super class of all classes that'll be contained in the DB.
+ * It implements all the abstractions needed by the remote storage to do all the queries, i.e.
+ * - The [dbPath] that precedes every key for a given class (e.g. "objectX/"),
+ * - The converter (to local) that converts a T DB object to a local StorableObject<T>,
+ * - The converter (to DB)]that converts an instance of a local StorableObject<T> to a DB T.
+ *
+ * It is the role of every subclass to define the path and the conversion methods.
+ *
+ * This class takes a generic type T as argument, corresponding to the data class that is
+ * going to be stored in the DB.
+ *
+ * @note attribute [key] should be the primary key used to store the object in the DB
+ *
+ * @note for the subclass to be correctly initialized, one instance should be created before any operation
+ */
+abstract class StorableObject<T : Any> (dbPath: String, private val key: String) {
+
+    init {
+        registerClassToCompanion(dbPath, this::class.toString(), ::toLocalObject)
+    }
```

Comment:
This should be in the init of the companion object instead: here it would be called at each instantiation of a `StorableObject`

Reply:
This should be called at the initialization of each of the subclasses, having that block in the companion object of `StorableObject` would call it once and thus would force the subclasses to have a companion object, we want to avoid this.

The block will be called at each instanciation of a subclass but this will only create low overhead and the same thing will be written each time so there is only a very low performance impact.

However if you have another solution, let me know :)

Reply:
This is indeed an acceptable trade-off !